azure/bastion/main.tf

```diff
@@ -72,8 +56,8 @@ resource "azurerm_network_security_group" "main" {
     source_port_range      = "*"
     destination_port_range = "443"
 
-    source_address_prefix      = var.allowed_cidrs == null ? "Internet" : null
-    source_address_prefixes    = var.allowed_cidrs != null ? var.allowed_cidrs : null
+    source_address_prefix      = length(local.whitelist) == 1 ? local.whitelist[0] : null
```

Comment:
Might be cleaner to run:

`one(local.whitelist)` instead of `local.whitelist[0]`

Reply:
I believe I tried a few things with this. One outcome was that it'd try to evaluate one(foo) despite the condition and error out as one(foo) needs the list to contain 0 or 1 items, which explains this outcome.

Reply:
Might be worth a comment in the code to explain this?